src/content/en/fundamentals/instant-and-offline/web-storage/cache-api.md

```diff
@@ -0,0 +1,248 @@
+parameter. If the named cache does not exist it will be created. This method returns a `Promise`
+that will resolve with the `Cache` object.
+
+    caches.open('my-cache).then((cache) => {
+      // do something with cache...
+    });
+
+## Retrieving from a cache
+
+To find an item in a cache, you can use the `match` method.
+
+    cache.match(request).then((response) => console.log(request, response));
+
+If `request` is a string it will first be converted to a `Request` by calling `new
+Request(request)`. The function returns a `Promise` that resolves to a `Response` if a matching
+entry is found, or `undefined` otherwise.
+
+To determine if two
+`Requests` match, more than just the URL is used. Two requests will be considered different
+if they have different query strings, `Vary` headers and/or methods (`GET`, `POST`, `PUT`, etc.).
+
+You can ignore some or all of these things by passing an options object as a second parameter.
+
+    const options = {
+      ignoreSearch: true,
+      ignoreMethod: true,
+      ignoreVary: true
+    };
+
+    cache.match(request, options).then(...);
+
+If more than one cached request matches then the one that was created first will be returned.
+
+If you want to retrieve *all* matching responses, you can use `cache.matchAll`.
+
+    const options = {
+      ignoreSearch: true,
+      ignoreMethod: true,
+      ignoreVary: true
+    };
+
+    cache.matchAll(request, options).then((responses) => {
+      console.log(`There are ${responses.length} matching responses.`);
+    });
+
+As a shortcut you can search over all caches at once by using `caches.match()` instead of calling
+`cache.match()` for each cache.
+
+## Searching
+
+The Cache API does not provide a way to search for requests or responses except for matching entries
+against a `Response` object. However, you can implement your own search using the features that are
+available.
+
+### Filtering
+
+One way is to iterate over all entries and filter down to the ones that you want. Let's say that you
+want to find all items that have URLs ending with '.png'.
+
+    async function findImages() {
+      // Get a list of all of the caches for this origin
+      const cacheNames = await caches.keys();
+      const result = [];
+
+      for (const name of cacheNames) {
+        // Open the cache
+        const cache = await caches.open(name);
+
+        // Get a list of entries. Each item will be a Request object
+        for (const request of await cache.keys()) {
+          // If the request URL matches, add the response to the result
+          if (request.url.endsWith('.png')) {
+            result.push(await cache.match(request));
+          }
+        }
+      }
+
+      return result;
+    }
+
+This way you can use any property of the `Request` and `Response` objects to filter the entries.
+Note that this will, of course, be slow if you search over large sets of data.
+
+### Creating an index
```

Comment:
Add a connector to the intro:

Another way to implement your own search is to maintain a separate index of...